fetch-nixpkgs.nix

```diff
@@ -0,0 +1,8 @@
+let
+  # temporary hack until scripts/nix-shell.sh ceases to use -p
```

Comment:
we could just skip using this hack and never use `-p` to begin with in this repo

Reply:
I simplified it significantly. It now assumes nix 2 is available and fetch-nixpkgs and fetchNixpkgs are a single file again.

Reply:
the whole `runCommand` and symlink part could also be removed, since thats only needed to allow you to `nix-build` the derivation and insert into `NIX_PATH`